core-tests/src/test/java/org/mule/runtime/core/internal/connection/PoolingConnectionManagementStrategyTestCase.java

```diff
@@ -142,10 +155,60 @@ public void failDueToInvalidConnection() throws ConnectionException {
 
   @Test(expected = ConnectionException.class)
   public void failDueToNullConnectionValidationResult() throws ConnectionException {
+    initStrategy();
+    connection1 = strategy.getConnectionHandler();
+    connection2 = strategy.getConnectionHandler();
+
     when(connectionProvider.validate(anyVararg())).thenReturn(null);
     strategy.getConnectionHandler().getConnection();
   }
 
+  @Test
+  public void defaultInitializationPolicyIsHonored() throws ConnectionException {
+    poolingProfile =
+        new PoolingProfile(5, -1, DEFAULT_MAX_POOL_WAIT, DEFAULT_POOL_EXHAUSTED_ACTION, DEFAULT_POOL_INITIALISATION_POLICY);
+    initStrategy();
+    verify(connectionProvider, times(1)).connect();
+    verify(connectionProvider, times(0)).disconnect(any());
+  }
+
+  @Test
+  public void initializationPolicyInitialiseAllIsHonored() throws ConnectionException {
+    poolingProfile = new PoolingProfile(5, -1, DEFAULT_MAX_POOL_WAIT, DEFAULT_POOL_EXHAUSTED_ACTION, INITIALISE_ALL);
+    initStrategy();
+    verify(connectionProvider, times(5)).connect();
+    verify(connectionProvider, times(0)).disconnect(any());
+  }
+
+  @Test
+  public void initializationPolicyInitialiseNoneIsHonored() throws ConnectionException {
+    poolingProfile = new PoolingProfile(5, -1, DEFAULT_MAX_POOL_WAIT, DEFAULT_POOL_EXHAUSTED_ACTION, INITIALISE_NONE);
+    initStrategy();
+    verify(connectionProvider, times(0)).connect();
+    verify(connectionProvider, times(0)).disconnect(any());
+  }
+
+  @Test
+  public void initializationPolicyIsLimitedByMaxIdle() throws ConnectionException {
+    poolingProfile = new PoolingProfile(5, 3, DEFAULT_MAX_POOL_WAIT, DEFAULT_POOL_EXHAUSTED_ACTION, INITIALISE_ALL);
+    initStrategy();
+    verify(connectionProvider, times(5)).connect();
```

Comment:
these two assertions are repeated all over the place. Extract to a method which receives two ints